tests/validation/tests/v3_api/test_configmaps.py

```diff
@@ -0,0 +1,489 @@
+        if key in testdata[0]['data']:
+            assert False
+        else:
+            assert True
+    if len(configmapdict.get('data')) == 0:
+        assert True
+
+    # Verify configmap is deleted by "kubectl get configmap" command
+    command = " get configmap " + configmap['name'] + " --namespace=" + ns.name
+    print("Command to obtain the configmap")
+    print(command)
+    result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+    print(result)
+
+    print("Verify that the configmap does not exist "
+          "and the error code returned is non zero ")
+    if result != 0:
+        assert True
+
+
+def create_and_validate_workload_with_configmap_as_volume(p_client, configmap,
+                                                          ns, keyvaluepair):
+    workload_name = random_test_name("test")
+    # Create Workload with configmap as volume
+    mountpath = "/test"
+    volumeMounts = [{"readOnly": False, "type": "volumeMount",
+                     "mountPath": mountpath, "name": "vol1"}]
+    con = [{"name": "test1",
+            "image": TEST_IMAGE,
+            "volumeMounts": volumeMounts}]
+
+    configmapname = configmap['name']
+
+    volumes = [{"type": "volume", "name": "vol1",
+                "configMap": {"type": "configMapVolumeSource",
+                              "defaultMode": 256,
+                              "name": configmapname,
+                              "optional": False}}]
+
+    workload = p_client.create_workload(name=workload_name,
+                                        containers=con,
+                                        namespaceId=ns.id, volumes=volumes)
+    validate_workload_with_configmap(p_client, workload, "deployment",
+                                     ns.name, keyvaluepair,
+                                     workloadwithconfigmapasvolume=True)
+
+    return workload
+
+
+def create_and_validate_workload_with_configmap_as_env_variable(p_client,
+                                                                configmap,
+                                                                ns,
+                                                                keyvaluepair):
+    workload_name = random_test_name("test")
+
+    # Create Workload with configmap as env variable
+    configmapname = configmap['name']
+
+    environmentdata = [{
+        "source": "configMap",
+        "sourceKey": None,
+        "sourceName": configmapname
+    }]
+    con = [{"name": "test",
+            "image": TEST_IMAGE,
+            "environmentFrom": environmentdata}]
+
+    workload = p_client.create_workload(name=workload_name,
+                                        containers=con,
+                                        namespaceId=ns.id)
+    validate_workload_with_configmap(p_client, workload, "deployment",
+                                     ns.name, keyvaluepair,
+                                     workloadwitconfigmapasenvvar=True)
+    return workload
```

Comment:
Should we delete the workload instead of returning it?

Reply:
Done